pkg/services/moving_expense/moving_expense_updater_test.go

```
@@ -0,0 +1,321 @@
+
+		originalMovingExpense := setupForTest(appCtx, nil, true)
+
+		updater := NewMovingExpenseUpdater()
+		contractedExpenseType := models.MovingExpenseReceiptTypeContractedExpense
+		rejectedStatus := models.PPMDocumentStatusRejected
+
+		expectedMovingExpense := &models.MovingExpense{
+			ID:                originalMovingExpense.ID,
+			MovingExpenseType: &contractedExpenseType,
+			Description:       models.StringPointer("Dumpster rental"),
+			PaidWithGTCC:      models.BoolPointer(true),
+			MissingReceipt:    models.BoolPointer(true),
+			Amount:            models.CentPointer(unit.Cents(67899)),
+			Status:            &rejectedStatus,
+			Reason:            models.StringPointer("Learn to recycle"),
+			SITStartDate:      models.TimePointer(time.Now()),
+			SITEndDate:        models.TimePointer(time.Now()),
+		}
+
+		updatedMovingExpense, updateErr := updater.UpdateMovingExpense(appCtx, *expectedMovingExpense, etag.GenerateEtag(originalMovingExpense.UpdatedAt))
+
+		suite.Nil(updateErr)
+		suite.Equal(originalMovingExpense.ID, updatedMovingExpense.ID)
+		suite.Equal(originalMovingExpense.DocumentID, updatedMovingExpense.DocumentID)
+		// filters out the deleted upload
+		suite.Len(updatedMovingExpense.Document.UserUploads, 1)
+		suite.Equal(*expectedMovingExpense.MovingExpenseType, *updatedMovingExpense.MovingExpenseType)
+		suite.Equal(*expectedMovingExpense.Description, *updatedMovingExpense.Description)
+		suite.Equal(*expectedMovingExpense.PaidWithGTCC, *updatedMovingExpense.PaidWithGTCC)
+		suite.Equal(*expectedMovingExpense.Amount, *updatedMovingExpense.Amount)
+		suite.Equal(*expectedMovingExpense.MissingReceipt, *updatedMovingExpense.MissingReceipt)
+		suite.Equal(*expectedMovingExpense.Status, *updatedMovingExpense.Status)
+		suite.Equal(*expectedMovingExpense.Reason, *updatedMovingExpense.Reason)
+		// Only the storage type receipt should be able to set these fields, would we rather reject
+		// the update outright than fail silently?
```

Rejecting might be nice, in that It would potentially help us catch errors where the UI doesn't clear out the values when switching from a storage type receipt to something else. It would also add clarity to the business rule that is happening on the post. (Ie. I could see someone looking at the API documentation and not realizing there is a business rule that prevents them from adding sit info to a non SIT expense.

---

I guess we aren't failing silently we just ignore those SIT field values if it's not type storage.  If they update the storage type from storage to something else then we nil out those fields in the updater https://github.com/transcom/mymove/pull/8957/files/e8ee85847d78da729041febb3274edd7c5e0d1dd#diff-f517feb1649debad15e67e21636e285955d87f62523e7ea50e04693de0871bc0R99-R104